### ISSUE TITLE
Set the 'immutable' Cache-Control directive for long-lived static assets

### ISSUE THREAD

**Issue by edmorley** [travis-ci/travis-ci#8601]
Firefox/Edge/Safari TP support a new Cache-Control `immutable` directive which prevents superfluous requests for long lived static assets.

Currently assets served from cdn.travis-ci.org are served with the header:
```
Cache-Control: public, max-age=31536000
```

This means that when say visiting https://travis-ci.org/profile/mozilla in Firefox/Edge and then hitting F5 (note, not ctrl+F5 for a force refresh - that intentionally always re-fetches everything), new HTTP 304s are seen for the fingerprinted assets that had a 1 year expiry and so did not need to be refetched, such as:
https://cdn.travis-ci.org/assets/vendor-8a29ab60ba44e825e9299cc7598ca2b0.css

Changing the reponse header to include `immutable` would prevent these unnecessary requests, ie:
```
Cache-Control: public, max-age=31536000, immutable
```

For further reading, see:
https://bitsup.blogspot.co.uk/2016/05/cache-control-immutable.html
https://hacks.mozilla.org/2017/01/using-immutable-caching-to-speed-up-the-web/
https://tools.ietf.org/html/draft-mcmanus-immutable-00
https://code.facebook.com/posts/557147474482256/this-browser-tweak-saved-60-of-requests-to-facebook/
https://www.keycdn.com/blog/cache-control-immutable/

(Moved from travis-ci/travis-web#1371)

**Comment by backspace** [comment]
It looks like your PRs change the caching on the `index.html` that links to the JS/CSS assets, rather than the caching on the assets themselves, does that seem accurate to you?

**Comment by edmorley** [comment]
Hi! Thank you for taking a look :-)

The PRs actually do the opposite of that, unless I'm overlooking something? The modified header is the header that is setting a max age of 1 year, rather than the max-age of zero (arguably the pre-existing `age` variable should be named `long_lived_asset_age` for clarity).

**Comment by backspace** [comment]
What I mean is that the JS and CSS are coming from cdn.travis-ci.org. The PRs are about changing the caching of the `index.html` file that’s served by `waiter`. Shouldn’t it be the asset caching that changes, not the index page?

**Comment by edmorley** [comment]
Ah I was presuming that cdn-travis-ci.org was being backed by the Waiter instance behind it. If not, then presumably this requires a change to a repo (or CDN dashboard) for which I'm not able to open a PR. (Either way, this doesn't modify the index page, it's at worst a no-op).

**Comment by backspace** [comment]
Yeah. I’ve looked a bit at our CDN setup but it wasn’t immediately obvious to me how to add this configuration. I’ll keep this open for now until I have more time to look into it, thanks for the suggestion.

**Comment by edmorley** [comment]
Still relevant:
```
$ curl -sfI https://cdn.travis-ci.org/assets/vendor-f5f3fdc40f949f46d471a91ac07a013e.js
| grep 'Cache-Control'
Cache-Control: public, max-age=31536000
```
